botocore/credentials.py

```diff
@@ -474,8 +705,79 @@ def _extract_creds_from_mapping(self, mapping, *key_names):
         return found
 
 
+class ProcessProvider(CredentialProvider):
```

Comment:
We should probably work to get a canonical name added for this provider (though I don't think it should block the merge).